### DISCUSSION TITLE
Traitement des groupes locaux

### DISCUSSION THREAD

**Discussion by jmthia** [LoicVeirman/HardenAD#129] [Q&A] [answered]
Bonjour,
Dans notre installation version 2022/07/28, les groupes locaux des différents serveurs sont modifiés automatiquement par une tâche planifiée. Cette installation est pris en charge par la tâche 140 (Import schedule tasks on the running system). 
Je ne trouve pas cette tâche dans la version 2.9.8. Est-ce que cette tâche est remplacée par la tâche 160 ? Si oui, le code est-il fonctionnel ?

Merci pour les réponses.
Cordialement,
Jean Marie

**Comment by LoicVeirman** [comment, marked as answer]
Bonjour,

Cette tâche vidait de façon régulière le contenu des groupes sensibles (par défaut, les groupes Admin Locaux Globaux pour les Tier 0, 1 et 2). elle n'est plus présente depuis la version 2.9.5.

Dans la version 2.9.8, cette fonctionnalité est apportée par un schedule ajouté via la GPO **HAD-TS-PDC-Flush-Admin-groups** et uniquement sur le PDC.

Si vous souhaitez mettre à jour le modèle, la version 2.10 va sortir courant septembre et contient de nombreuses amélioration. Il est possible également que je vous accompagne pour faire une "monté de version" depuis votre version actuelle en vous écrivant directement un script d'update.

**Comment by jmthia** [comment]
Bonjour Loïc,
Merci pour la réponse, qui entraine une autre question plus générale sur les groupes locaux des serveurs. Dans la version 2022, le fait de mettre un serveur dans l'OU T1\Servers, ajoutait dans le groupe administrateurs du serveur les groupes L-S-T1_LocalAdmins_HostName et L-S-T1_LocalAdmins_Serveurs. Si j'ai bien compris les explications, ces groupes groupe via une GPO assurent le modèle de connexion en tier.
En 2.9.8, ces groupes ne sont pas alimentés automatiquement.  Quel est la séquence qui gère cette fonctionnalité ?
Par avance merci.

**Reply by LoicVeirman** [reply]
Ces groupes sont toujours présents, via les GPO localAdmins - en deux actions distinctes (une pour le global et une pour le spécifique), ce qui évite l'erreur à l'application si le groupe n'existe.

Le groupe de chaque machine est automatiquement créé par un schedule sur chaque DC (détection sur event, ce dernier étant déclencher sur le DC qui traite la demande de création). Il est positionné dans l'OU LocalAdmins de son Tier et, en cas de non identification du Tier, dans celle du Tier 0.

Si un objet Ordinateur est déplacé, le groupe qui lui est associé sera challengé et au besoin délocalisé dans la bonne OU de son nouveau tier.

Si un objet Ordinateur est supprimé, le groupe l'es également.

Enfin, deux tâches tournent la nuit pour réconcilier les éventuels loupés à la détection.

Les groupes sont volontairement vide dès le départ. Ils sont vidés en cas de changement de Tier.

**Reply by jmthia** [reply]
Les groupes locaux ne sont pas alimentés chez moi (v2.9.8), est-ce qu'il faut activer la séquence 160 qui a été désactivé "for rewrite in 2.9.9" ?
vu le commentaire, je ne l'ai activé. c'est peut-être ce qui me manque.

**Reply by LoicVeirman** [reply]
"Les groupes locaux ne sont pas alimentés chez moi (v2.9.8)" : vérifiez le schedule sur les DC. Attention, reboot des DC obligatoires sinon le code n'exécute pas l'installation.

**Comment by jmthia** [comment]
Par ailleurs, je remarque que les groupes localAdmins  des serveurs ne sont plus préfixés avec le tier L-S-LocalAdmins_srvAD comparative ment à L-S-T0_LocalAdmins_Servers.

**Reply by LoicVeirman** [reply]
La notion de tier n'a pas de sens pour le local admin, mais vous pouvez adapter le code pour le remettre.
